cms/djangoapps/course_creators/admin.py

```diff
@@ -30,26 +33,53 @@ def get_email(obj):
 get_email.short_description = 'email'
 
 
+class CourseCreatorForm(forms.ModelForm):
+    """
+    Admin form for course creator
+    """
+    class Meta:
+        model = CourseCreator
+        fields = '__all__'
+
+    def clean(self):
+        """
+        Running validation of the form. Validation to stop user from entering
+        false data.
+        """
+        all_orgs = self.cleaned_data.get("all_organizations")
+        updated_state = self.cleaned_data.get("state")
+        orgs = self.cleaned_data.get("orgs").exists()
+        add_role = updated_state == CourseCreator.GRANTED
+        if not add_role and orgs:
+            raise ValidationError("Organizations cannot be added if the role is not granted.")
```

Comment:
@farhaanbukhsh, this causes issues when a request is initially granted for a specific set of organizations and then the admin tries to change the state to any state other than `CourseCreator.GRANTED`. For example, this will happen when the request is denied after granting it.

Reply:
I have removed this validation and instead, when any of the statuses is changed from Granted to any other I remove all the orgs associated with it.